control-plane-operator/hostedclusterconfigoperator/controllers/resources/globalps/pullsecret.go

```diff
@@ -0,0 +1,297 @@
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/var/lib/kubelet",
+									Type: ptr.To(corev1.HostPathDirectory),
+								},
+							},
+						},
+						{
+							Name: "dbus",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/var/run/dbus",
+								},
+							},
+						},
+					},
+				},
+			},
+			UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+				Type: appsv1.RollingUpdateDaemonSetStrategyType,
+				RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+					MaxUnavailable: &maxUnavailable,
+					MaxSurge:       &maxSurge,
+				},
+			},
+		}
+
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to create global pull secret daemon set: %w", err)
+	}
+
+	return nil
+}
+
+func ValidateUserProvidedPullSecret(pullSecret *corev1.Secret) ([]byte, error) {
+	var dockerConfigJSON credentialprovider.DockerConfigJSON
+
+	// Validate that the pull secret contains the dockerConfigJson key
+	if _, ok := pullSecret.Data[corev1.DockerConfigJsonKey]; !ok {
+		return nil, fmt.Errorf("pull secret data is not a valid docker config json")
+	}
+
+	// Validate that the pull secret is a valid DockerConfigJSON
+	pullSecretBytes := pullSecret.Data[corev1.DockerConfigJsonKey]
+	if err := json.Unmarshal(pullSecretBytes, &dockerConfigJSON); err != nil {
+		return nil, fmt.Errorf("invalid docker config json format: %w", err)
+	}
+
+	// Validate that the pull secret contains at least one auth entry
+	if len(dockerConfigJSON.Auths) == 0 {
+		return nil, fmt.Errorf("docker config json must contain at least one auth entry")
+	}
+
+	return pullSecretBytes, nil
+}
+
+// MergePullSecrets merges two pull secrets into a single pull secret.
+// The additional pull secret is merged with the original pull secret.
+// If an auth entry already exists, it will be overwritten.
+// The resulting pull secret is returned as a JSON string.
+// Not using credentialprovider.DockerConfigJSON because it does not support
+// marshaling the auth field.
+func MergePullSecrets(ctx context.Context, originalPullSecret, userProvidedPullSecret []byte) ([]byte, error) {
+	var (
+		originalAuths         map[string]any
+		userProvidedAuths     map[string]any
+		originalJSON          map[string]any
+		userProvidedJSON      map[string]any
+		globalPullSecretBytes []byte
+		err                   error
+	)
+
+	// Unmarshal original pull secret
+	if err = json.Unmarshal(originalPullSecret, &originalJSON); err != nil {
+		return nil, fmt.Errorf("invalid original pull secret format: %w", err)
+	}
+	originalAuths = originalJSON["auths"].(map[string]any)
+
+	// Unmarshal additional pull secret
+	if err = json.Unmarshal(userProvidedPullSecret, &userProvidedJSON); err != nil {
+		return nil, fmt.Errorf("invalid user provided pull secret format: %w", err)
+	}
+	userProvidedAuths = userProvidedJSON["auths"].(map[string]any)
+
+	// Merge auths
+	for k, v := range userProvidedAuths {
+		originalAuths[k] = v
```

Comment:
Just want to confirm that with merge, user provided pull secret may override the global pull secret, is it what we want ? I am asking because I see a
```
This feature is to provide a mechanism where the customer can place a pull secret in cluster which would then result in it being appended to the global pull secret
```
in the Jira issue description which is saying `append`.